### PR TITLE
pkg/identity: create identity package

### PR DESCRIPTION
**Pull request by ianvernon**
Factor out code from policy package related to identity allocation, types, etc.
into a separate package. This was motivated by cyclic import issues faced in
PR #2875. Update code to use this package accordingly. No change in
functionality should occur as part of this commit.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

File: pkg/identity/allocator.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
+package identity
```
Comment (aanm) [inline pkg/identity/allocator.go:15]: missing header

File: pkg/identity/identity.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package identity
```
Comment (aanm) [inline pkg/identity/identity.go:15]: missing header

**Review by aanm — DISMISSED** [review]

**Comment by aanm** [comment]
Should we run a nightly test?

**Comment by ianvernon** [comment]
@aanm good idea. I'll run one after I push changes which address your comments.

File: pkg/identity/identity.go (hunk 2 above)
Comment (ianvernon) [reply pkg/identity/identity.go:15]: Fixed.

File: pkg/identity/allocator.go (hunk 1 above)
Comment (ianvernon) [reply pkg/identity/allocator.go:15]: Fixed.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-nightly

File: pkg/identity/identity.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/labels"
+)
+
+// Identity is the representation of the security context for a particular set of
+// labels.
+type Identity struct {
+	// Identity's ID.
+	ID NumericIdentity `json:"id"`
+	// Set of labels that belong to this Identity.
```
Comment (rlenglet) [inline pkg/identity/identity.go:27]:
```
Labels is the sorted list of labels that is unique to this Identity.
```

**Review by rlenglet — APPROVED** [review]

**Review by manalibhutiyani — APPROVED** [review]

**Review by joestringer — APPROVED** [review]
LGTM from CLI perspective.

**Comment by ianvernon** [comment]
Ginkgo tests look like they failed due to timeout. I suspect that this is due to the Envoy Rebase being merged and causing tests to run longer than expected. I'll look into fixing that.

**Comment by ianvernon** [comment]
Looked into what parameters the build was launched with; despite master containing an updated Jenkinsfile timeout of 90 mins for the `BDD-Test` stage, the Jenkins job does not appear to pull in Jenkinsfile changes from master into branches despite rebasing against master. I suspect this is a chicken-and-egg problem (Jenkinsfile has to be read before the build can launch / rebase can occur), so I'll rebase and relaunch the tests after the nightly tests finish up.

**Comment by ianvernon** [comment]
Nightly test failed due to default deny on K8s 1.7 which is a known issue / was merged in a recent PR after I posted this one. Will rebase and re-run non-nightly tests again.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**